### PR TITLE
feat(ia): add settings seo 

### PR DESCRIPTION
**Pull request by jaredrethman**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

- Add SEO wizard to Settings SEO page
- Add new hook for simple text input validation
- Add robust FE form field validation

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets i.e. `git checkout origin/feat/ia-settings-seo && npm run build`.
2. Navigate to _/wp-admin/admin.php?page=newspack-settings#/seo_
3. Observe all SEO input fields are present.

Screenshot 2024-08-23 at 09 54 40

4. Test Verification codes. Ensure values are stored and appropriate error displays when below rule(s) are violated.
   - Google rules: alpha-numeric, hyphen and underscore chars
   - Bing rules: aA-fF chars, hyphen and underscore chars
5. Test Accounts. Ensure values are stored and appropriate error displays when below rule(s) are violated.
   - X rules: Alpha numeric chars and underscores. [Max length is 15](https://charactercount.tools/twitter/#:~:text=10000%20characters-,Twitter%20handle%20max%20length,-15%20characters).
   - The rest of the account fields require a valid url.

Example of errors:
Screenshot 2024-08-23 at 10 11 28

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/wizards/newspack/views/settings/seo/verification-codes.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	return (
+		<Grid>
+			<TextControl
+				label="Google"
+				onChange={ ( google: string ) => setData( { ...data, google } ) }
+				value={ data.google }
+				help={
+					<Fragment>
+						{ __( 'Get your verification code in', 'newspack' ) + ' ' }
+						<ExternalLink href="https://www.google.com/webmasters/verification/verification?tid=alternate">
```
Comment (dkoo) [inline src/wizards/newspack/views/settings/seo/verification-codes.tsx]: Looks like the URL for this settings page has changed. It's now `https://search.google.com/search-console/ownership?resource_id=<url>` where `<url>` is the encoded URL for the site, e.g.

`https://search.google.com/search-console/ownership?resource_id=https%3A%2F%2Fnewspack.com%2F`

File: src/wizards/newspack/views/settings/seo/constants.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * 2. Field label.
+ * 3. Field placeholder.
+ * 4. (Optional) Validation callback name.
+ * 5. (Optional) Field error message.
+ */
+export const ACCOUNTS = [
+	[
+		'twitter',
+		__( 'X (formerly Twitter) Handle', 'newspack-plugin' ),
+		'username',
```
Comment (dkoo) [inline src/wizards/newspack/views/settings/seo/constants.ts]: Should this be a translatable string?

File: src/wizards/newspack/views/settings/seo/verification-codes.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+			/>
+			<TextControl
+				label="Bing"
+				onChange={ ( bing: string ) => setData( { ...data, bing } ) }
+				value={ data.bing }
+				help={
+					<Fragment>
+						{ `${ __( 'Get your verification code in', 'newspack' ) } ` }
+						<ExternalLink href="https://www.bing.com/toolbox/webmaster/#/Dashboard/">
+							{ __( 'Bing Webmaster Tool', 'newspack' ) }
```
Comment (dkoo) [inline src/wizards/newspack/views/settings/seo/verification-codes.tsx]:
```suggestion
							{ __( 'Bing Webmaster Tools', 'newspack' ) }
```

Just based on their page title:

Screenshot 2024-09-06 at 5 20 28 PM

**Review by dkoo — APPROVED** [review]
Pre-approving since the only feedback is updating some strings!

**Comment by jaredrethman** [comment]
@dkoo appreciate the review, I have committed the string changes you recommended.

File: src/wizards/newspack/views/settings/seo/constants.ts (hunk 2 above)
Comment (jaredrethman) [reply src/wizards/newspack/views/settings/seo/constants.ts]: Indeed, have updated

File: src/wizards/newspack/views/settings/seo/verification-codes.tsx (hunk 3 above)
Comment (jaredrethman) [reply src/wizards/newspack/views/settings/seo/verification-codes.tsx]: Nice catch, updated!

**Pull request merged**